src/server.py

```diff
@@ -90,7 +90,13 @@ def pause_with_fade_out(self):
         else:
             self._pending_state_change = Gst.State.PAUSED
             # The element will be set to PAUSED state when volume reaches 0.
-            self._add_fade_out(volume_elem, self.fade_out)
+            try:
+                self._add_fade_out(volume_elem, self.fade_out)
+            except ValueError as ex:
+                self.logger.warning("Fade out effect could not be applied. "
+                                    "Pausing.")
+            self.pipeline.set_state(Gst.State.PAUSED)
```

Comment:
Maybe you meant to indent these two lines to be part of the except clause? It seems like this will pause the sound before the fade out starts, instead of when it ends.

Reply:
yes